rholang/src/main/scala/coop/rchain/rholang/interpreter/match.scala

```diff
@@ -292,14 +270,26 @@ object SpatialMatcher {
         forcedYield <- foldMatch(target.data, pattern.data, (t: Par, p: Par) => spatialMatch(t, p))
       } yield forcedYield
 
-  /**
-    * Note that currently there should be no way to put a GPrivate in a pattern
-    * because patterns start with an empty environment.
-    * We're going to write the obvious definition anyway.
-    */
-  def spatialMatch(target: GPrivate, pattern: GPrivate): OptionalFreeMap[Unit] =
-    if (target == pattern)
-      StateT.pure(Unit)
+  def spatialMatch(target: Receive, pattern: Receive): OptionalFreeMap[Unit] =
+    if (target.persistent != pattern.persistent)
+      StateT.liftF(None)
+    else
+      for {
+        _ <- listMatchSingle[ReceiveBind](target.binds,
+                                          pattern.binds,
+                                          spatialMatch,
+                                          (p, rb) => p,
+                                          None,
+                                          false)
+        _ <- spatialMatch(target.body.get, pattern.body.get)
+      } yield Unit
```

Comment:
I would switch the if/else case so that we use == instead of !=

Reply:
I think it's easier to read if the short case goes first.